### ISSUE TITLE
Bug: Workspace + BSR dependencies leads to failing BSR dependency import

### ISSUE THREAD

**Issue by mcquackers** [bufbuild/buf#975]
Example:
https://github.com/mcquackers/buf-petstore

OS: MacOS Monterey 12.1
Buf Version: v1.0.0

There is a directory organization with a workspace file that seems to break BSR remote dependencies.
That organization seems to be:
a/b/v_n_/file.proto
z/a/c/v_n_/other_file.proto

where the buf.work.yaml lists __a__ and __z__ in __directories__

**Comment by amckinney** [comment]
I checked out your repository (the broken commit in particular) and fixed the problem.

You've set up your workspace incorrectly - you need to specify dependencies (via a `buf.lock`) for each module in the workspace that has dependencies. In this case, you need the `buf.build/googleapis/googleapis` dependency in your `townville` module, so your file tree should be set up like so:

```sh
.
├── buf.work.yaml
├── submodule
│   └── townville
│       └── currency
│           └── v1
│               └── currency.proto
└── townville
    ├── buf.lock
    ├── buf.yaml
    └── petstore
        ├── buf.gen.yaml
        └── v1
            └── petstore.proto
```

With that said, I'd also move the `buf.gen.yaml` file up alongside the `buf.work.yaml` since that's aligned with our best practices (as described in the [tour](https://docs.buf.build/tour/introduction)).

Please let us know if you have any other questions!

**Comment by mcquackers** [comment]
@amckinney thanks for pointing out the issue!  I figured `buf.yaml` and `buf.lock` would belong in the `v1` as dependencies might differ between v1/v2/v3, etc.

**Comment by amckinney** [comment]
On that note, check out the documentation on versioning best practices [here](https://docs.buf.build/best-practices/module-development#package-versions). That should help clear things up a bit.

TL;DR - you still want to version individual packages within each of your modules.
